pkg/imageverifycache/client.go

```diff
@@ -75,19 +101,29 @@ func (c *cache) Set(ctx context.Context, policy kyvernov1.PolicyInterface, ruleN
 	if !c.isCacheEnabled {
 		return false, nil
 	}
-	c.logger.Info("Successfully set cache", "policy", policy.GetName(), "ruleName", ruleName, "imageRef", imageRef)
+	key := generateKey(policy, ruleName, imageRef)
+
+	stored := c.Cache.SetWithTTL(key, nil, 1, c.ttl)
+	if stored {
+		c.logger.Info("Successfully set cache", "policy", policy.GetName(), "ruleName", ruleName, "imageRef", imageRef)
+		return true, nil
+	}
 	return false, nil
 }
 
 func (c *cache) Get(ctx context.Context, policy kyvernov1.PolicyInterface, ruleName string, imageRef string) (bool, error) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
-
 	c.logger.Info("Searching in cache", "policy", policy.GetName(), "ruleName", ruleName, "imageRef", imageRef)
 	if !c.isCacheEnabled {
 		return false, nil
 	}
+	key := generateKey(policy, ruleName, imageRef)
+	_, found := c.Cache.Get(key)
+	if found {
+		c.logger.Info("Cache entry found", "policy", policy.GetName(), "ruleName", ruleName, "imageRef", imageRef)
```

Comment:
policy namespace is interesting too.
and you could create a local logger and use `WithValues` instead of repeating all kv pairs every time.

Reply:
sure, I will do that.

Reply:
done